src/server/active_tran_server.cpp

```diff
@@ -69,28 +110,21 @@ int active_tran_server::init_page_server_hosts (const char *db_name)
       return ER_HOST_PORT_PARAMETER;
     }
 
-  long port = -1;
-  try
-    {
-      port = std::stol (hosts.substr (col_pos + 1));
-    }
-  catch (...)
-    {
-    }
+  size_t pos = 0;
+  std::string delimiter = ",";
+  bool connected;
+  int exit_code = ER_HOST_PORT_PARAMETER;
 
-  if (port < 1 || port > USHRT_MAX)
+  while ((pos = hosts.find (delimiter)) != std::string::npos)
     {
-      er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, ER_HOST_PORT_PARAMETER, 2, prm_get_name (PRM_ID_PAGE_SERVER_HOSTS),
-	      hosts.c_str ());
-      return ER_HOST_PORT_PARAMETER;
-    }
-  m_ps_port = port;
+      std::string token = hosts.substr (0, pos);
+      hosts.erase (0, pos + delimiter.length());
 
-  // host and port seem to be OK
-  m_ps_hostname = hosts.substr (0, col_pos);
-  er_log_debug (ARG_FILE_LINE, "Page server hosts: %s port: %d\n", m_ps_hostname.c_str (), m_ps_port);
+      exit_code = std::max (parse_server_host (token, db_name, &connected), exit_code);
+    }
 
-  return connect_to_page_server (m_ps_hostname, m_ps_port, db_name);
+  exit_code = std::max (parse_server_host (hosts, db_name, &connected), exit_code);
+  return exit_code;
```

Comment:
The max exit code is a trick that may confuse future readers. Let me suggest an alternative handling.

The issue says to ignore the invalid hosts and to the connect to the first valid host. There is a valid host if connection node list is not empty:

```c
  while ((pos = hosts.find (delimiter)) != std::string::npos)
    {
      std::string token = hosts.substr (0, pos);
      hosts.erase (0, pos + delimiter.length());

      parse_server_host (token, db_name);
    }
  if (connection_list.empty ())
    {
      // no valid hosts
      ASSERT_ERROR_AND_SET (exit_code); // er_set was called
      return exit_code;
    }
  return connect_to_page_server (connection_list.front (), db_name);
```

I also suggest to change connect_to_page_server signature to use cubcomm::node.